### PR TITLE
Handle local file URI without authority

### PR DESCRIPTION
**Pull request by SaravananSathyanandhaQC**
The `fsspec` Local implementation currently doesn't handle file URIs that skip the authority properly, as per https://www.rfc-editor.org/rfc/rfc8089#appendix-A - ref: https://github.com/fsspec/universal_pathlib/issues/81

(The pytest CI seems to hang on my fork, not certain why)

### CODE REVIEW

File: fsspec/implementations/tests/test_local.py [hunk 1]
```diff
@@ -628,6 +628,13 @@ def test_strip_protocol_expanduser():
     assert not LocalFileSystem._strip_protocol("./").endswith("/")
 
 
+def test_strip_protocol_no_authority():
+    path = "file:\\foo\\bar" if WIN else "file:/foo/bar"
+    stripped = LocalFileSystem._strip_protocol(path)
+    assert "file:" not in stripped
+    assert stripped.endswith("/foo/bar")
```
Comment (martindurant) [inline fsspec/implementations/tests/test_local.py:635]: Why not `== "/foo/bar"`, is there any other possibility?

Comment (SaravananSathyanandhaQC) [reply fsspec/implementations/tests/test_local.py:635]: Because on Windows the absolute path includes a drive letter so becomes `C:/foo/bar`

Comment (SaravananSathyanandhaQC) [reply fsspec/implementations/tests/test_local.py:635]: See failed windows run https://github.com/SaravananSathyanandhaQC/filesystem_spec/actions/runs/4530930619/jobs/7980467557

Comment (martindurant) [reply fsspec/implementations/tests/test_local.py:635]: Got it, thank you.

**Comment by martindurant** [comment]
Seems like CI is having problems generally... we will investigate.

**Comment by martindurant** [comment]
Now github itself is having issues, so we wait :)
https://www.githubstatus.com/

**Comment by SaravananSathyanandhaQC** [comment]
The green tick of joy doesn't show here for some reason, but it's showing on my fork

**Pull request merged**